src/ConfigWidgets/SymbolsDialog.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ConfigWidgets/SymbolsDialog.h"
+
+#include <qpushbutton.h>
+
+#include <QFileDialog>
+#include <QPushButton>
+#include <algorithm>
+#include <memory>
+
+#include "OrbitBase/Logging.h"
+#include "ui_SymbolsDialog.h"
+
+namespace orbit_config_widgets {
+
+SymbolsDialog::~SymbolsDialog() noexcept = default;
+
+SymbolsDialog::SymbolsDialog(QWidget* parent)
+    : QDialog(parent), ui_(std::make_unique<Ui::SymbolsDialog>()) {
+  ui_->setupUi(this);
+
+  QObject::connect(ui_->addButton, &QPushButton::clicked, this, [this]() {
+    QString directory = QFileDialog::getExistingDirectory(this, "Select Symbol Directory");
+    if (!directory.isEmpty()) {
+      AddSymbolPath(std::filesystem::path{directory.toStdString()});
+    }
```

Comment:
What if the directory is already part of the list? Should we deduplicate?

Reply:
I extended the AddSymbolsPath function to an TryAddSymbolsPath, which only allows adding new paths. ptal